packages/plugin-commands-patching/test/patch.test.ts

```diff
@@ -94,3 +94,82 @@ describe('patch and commit', () => {
       .rejects.toThrow(`The target directory already exists: '${editDir}'`)
   })
 })
+
+describe('patching should work when there is a no EOL in the patched file', () => {
+  let defaultPatchOption: patch.PatchCommandOptions
+
+  beforeEach(() => {
+    prepare({
+      dependencies: {
+        'safe-execa': '0.1.2',
+      },
+    })
+
+    const cacheDir = path.resolve('cache')
+    const storeDir = path.resolve('store')
+
+    defaultPatchOption = {
+      cacheDir,
+      dir: process.cwd(),
+      pnpmHomeDir: '',
+      rawConfig: {
+        registry: `http://localhost:${REGISTRY_MOCK_PORT}/`,
+      },
+      registries: { default: `http://localhost:${REGISTRY_MOCK_PORT}/` },
+      storeDir,
+      userConfig: {},
+    }
+  })
+
+  test('adding content in newline should works fine', async () => {
+    const output = await patch.handler(defaultPatchOption, ['safe-execa@0.1.2'])
+    const userPatchDir = output.substring(output.indexOf(':') + 1).trim()
+    const tempDir = os.tmpdir()
+
+    expect(userPatchDir).toContain(tempDir)
+    expect(fs.existsSync(userPatchDir)).toBe(true)
+    expect(fs.existsSync(path.join(userPatchDir, 'lib/index.js'))).toBe(true)
+
+    fs.appendFileSync(path.join(userPatchDir, 'lib/index.js'), '\n// test patching', 'utf8')
+
+    await patchCommit.handler({
+      ...DEFAULT_OPTS,
+      dir: process.cwd(),
+    }, [userPatchDir])
+
+    const { manifest } = await readProjectManifest(process.cwd())
+    expect(manifest.pnpm?.patchedDependencies).toStrictEqual({
+      'safe-execa@0.1.2': 'patches/safe-execa@0.1.2.patch',
+    })
+    const patchContent = fs.readFileSync('patches/safe-execa@0.1.2.patch', 'utf8')
+    expect(patchContent).toContain('diff --git')
+    expect(patchContent).toContain('// test patching')
+    expect(patchContent).not.toContain('No newline at end of file')
+    expect(fs.readFileSync('node_modules/safe-execa/lib/index.js', 'utf8')).toContain('// test patching')
+  })
+
+  test('append content should works fine', async () => {
+    const output = await patch.handler(defaultPatchOption, ['safe-execa@0.1.2'])
+    const userPatchDir = output.substring(output.indexOf(':') + 1).trim()
+    const tempDir = os.tmpdir()
+
+    expect(userPatchDir).toContain(tempDir)
+    expect(fs.existsSync(userPatchDir)).toBe(true)
+    expect(fs.existsSync(path.join(userPatchDir, 'lib/index.js'))).toBe(true)
+
+    fs.appendFileSync(path.join(userPatchDir, 'lib/index.js'), '// patch without newline', 'utf8')
+
+    await patchCommit.handler({
+      ...DEFAULT_OPTS,
+      dir: process.cwd(),
+    }, [userPatchDir])
+
+    const { manifest } = await readProjectManifest(process.cwd())
+    expect(manifest.pnpm?.patchedDependencies).toStrictEqual({
+      'safe-execa@0.1.2': 'patches/safe-execa@0.1.2.patch',
+    })
+    const patchContent = fs.readFileSync('patches/safe-execa@0.1.2.patch', 'utf8')
+    expect(patchContent).toContain('No newline at end of file')
```

Comment:
Sorry, I didn't understand the unit test code here, can you explain it? 🙇‍♂️

In my personal understanding, the behavior of a single test is as follows.

### Case 1:

#### Input
```diff
diff --git a/1 b/1
index 78981922..dba81ac2 100644
--- a/1
+++ b/1
@@ -1 +1 @@
-a
+ax
\ No newline at end of file
```

#### Output
```diff
diff --git a/1 b/1
index 78981922..dba81ac2 100644
--- a/1
+++ b/1
@@ -1 +1 @@
-a
+ax
```

### Case 2:

#### Input
```diff
diff --git a/1 b/1
index 78981922..dba81ac2 100644
--- a/1
+++ b/1
@@ -1 +1 @@
-a
- \ No newline at end of file
+ax
\ No newline at end of file
```

#### Output
```diff
diff --git a/1 b/1
index 78981922..dba81ac2 100644
--- a/1
+++ b/1
@@ -1 +1 @@
-a
- \ No newline at end of file
+ax
```

Reply:
In case 2, although there are two lines `\ No newline at end of file`, we only remove the last one, but the final patch result is still work fine, case 2 should be able to cover the case of  `\ No newline at end of file` that happens to exist in the user code (except in the last line)